From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Sun, 22 Mar 2026 00:46:24 +0800
Subject: [PATCH] H.BUILD: MindustryX group for publish

---
 build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build.gradle b/build.gradle
index 158a38eb9c1bd4c61a8791dedb63fa0ae9608d01..728978cea5af6f9994fc304783cc1071a1fb3453 100644
--- a/build.gradle
+++ b/build.gradle
@@ -63,7 +63,7 @@ libraries.lwjgl = [
 
 allprojects{
     apply plugin: 'maven-publish'
-    group = 'com.github.Anuken'
+    group = 'com.github.TinyLake.MindustryX'
     version = '1.0'
 
     repositories{
