--- ./mindustry/gen/Call.java
+++ ./mindustry/gen/Call.java
@@ -32,6 +32,7 @@ import mindustry.ui.builder.MenuResult;
 import mindustry.ui.builder.UiBuilder;
 import mindustry.world.Block;
 import mindustry.world.Tile;
+import mindustryX.features.*;
 
 public class Call {
   public static void adminRequest(Player other, Packets.AdminAction action, Object params) {
@@ -318,6 +319,7 @@ public class Call {
   }
 
   public static void clientPlanSnapshot(int groupId, TypeIO.ClientBuildPlans plans) {
+    if(LogicExt.mockProtocol<156)return;
     if(mindustry.Vars.net.client()) {
       ClientPlanSnapshotCallPacket packet = new ClientPlanSnapshotCallPacket();
       packet.groupId = groupId;
@@ -1417,6 +1419,7 @@ public class Call {
   }
 
   public static void pingLocation(Player player, float x, float y, String text) {
+    if(LogicExt.mockProtocol<156)return;
     if(mindustry.Vars.net.server() || !mindustry.Vars.net.active()) {
       mindustry.input.InputHandler.pingLocation(player, x, y, text);
     }
--- ./mindustry/gen/ClientSnapshotCallPacket.java
+++ ./mindustry/gen/ClientSnapshotCallPacket.java
@@ -9,6 +9,7 @@ import mindustry.net.NetConnection;
 import mindustry.net.Packet;
 import mindustry.world.Block;
 import mindustry.world.Tile;
+import mindustryX.features.*;
 
 public class ClientSnapshotCallPacket extends Packet {
   private byte[] DATA = NODATA;
@@ -82,8 +83,10 @@ public class ClientSnapshotCallPacket extends Packet {
     WRITE.bool(shooting);
     WRITE.bool(chatting);
     WRITE.bool(building);
-    mindustry.io.TypeIO.writeBlock(WRITE, selectedBlock);
-    WRITE.i(selectedRotation);
+    if(LogicExt.mockProtocol>=155){
+        mindustry.io.TypeIO.writeBlock(WRITE, selectedBlock);
+        WRITE.i(selectedRotation);
+    }
     mindustry.io.TypeIO.writePlansQueueNet(WRITE, plans);
     WRITE.f(viewX);
     WRITE.f(viewY);
--- ./mindustry/gen/Player.java
+++ ./mindustry/gen/Player.java
@@ -1155,15 +1155,17 @@ public class Player implements Drawc, Entityc, IndexableEntity__draw, IndexableE
       read.f();
     }
     this.name = mindustry.io.TypeIO.readString(read);
-    if(!islocal) {
-      this.selectedBlock = mindustry.Vars.content.getByID(mindustry.ctype.ContentType.block, read.s());
-    } else {
-      mindustry.Vars.content.getByID(mindustry.ctype.ContentType.block, read.s());
-    }
-    if(!islocal) {
-      this.selectedRotation = read.i();
-    } else {
-      read.i();
+    if(LogicExt.mockProtocol>=155){
+        if(!islocal) {
+            this.selectedBlock = mindustry.Vars.content.getByID(mindustry.ctype.ContentType.block, read.s());
+        } else {
+            mindustry.Vars.content.getByID(mindustry.ctype.ContentType.block, read.s());
+        }
+        if(!islocal) {
+            this.selectedRotation = read.i();
+        } else {
+            read.i();
+        }
     }
     if(!islocal) {
       this.shooting = read.bool();

