From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Fri, 24 May 2024 19:45:32 +0800
Subject: [PATCH] FC(LogicExt) limitUpdate

---
 core/src/mindustry/entities/EntityGroup.java | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/core/src/mindustry/entities/EntityGroup.java b/core/src/mindustry/entities/EntityGroup.java
index f315ac8c950741e657c692d00442a022f9daf0fb..ea468bf62ac1f08c1f8fe56043089d55b51932a0 100644
--- a/core/src/mindustry/entities/EntityGroup.java
+++ b/core/src/mindustry/entities/EntityGroup.java
@@ -99,6 +99,17 @@ public class EntityGroup<T extends Entityc> implements Iterable<T>{
             DebugUtil.traceEnd();
             return;
         }
+        if(LogicExt.limitUpdate.get() > 0 && LogicExt.limitUpdateTimer != 0){
+            Core.camera.bounds(viewport);
+            viewport.grow(LogicExt.limitUpdate.get() * 2);
+            for(index = 0; index < array.size; index++){
+                Entityc e = array.items[index];
+                if(e instanceof Position p && !viewport.contains(p.getX(), p.getY()))
+                    continue;
+                array.items[index].update();
+            }
+            return;
+        }
         for(index = 0; index < array.size; index++){
             array.items[index].update();
         }
