From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Sun, 15 Sep 2024 15:53:45 +0800
Subject: [PATCH] FC: reliableSync

---
 core/src/mindustry/net/Net.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/src/mindustry/net/Net.java b/core/src/mindustry/net/Net.java
index a4a580ba10bac5bdb7116df6cd451527ed19685b..ae3a07a5ad2e0d7a05cd3d73ab0242415a0f8bb5 100644
--- a/core/src/mindustry/net/Net.java
+++ b/core/src/mindustry/net/Net.java
@@ -250,6 +250,7 @@ public class Net{
         if(server){
             provider.sendAllServer(object, reliable);
         }else{
+            reliable |= LogicExt.reliableSync.get();
             provider.sendClient(object, reliable);
         }
     }
