From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Sun, 15 Sep 2024 15:53:45 +0800
Subject: [PATCH] FC: reliableSync

---
 core/src/mindustry/net/Net.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/src/mindustry/net/Net.java b/core/src/mindustry/net/Net.java
index f13f739f44c6a0be7fabd47e9de236c2722d7c79..c1b89692968a21c682fedd807d7c57887259637f 100644
--- a/core/src/mindustry/net/Net.java
+++ b/core/src/mindustry/net/Net.java
@@ -251,6 +251,7 @@ public class Net{
         if(server){
             provider.sendAllServer(object, reliable);
         }else{
+            reliable |= LogicExt.reliableSync.get();
             provider.sendClient(object, reliable);
         }
     }
