From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Sat, 4 Apr 2026 00:40:01 +0800
Subject: [PATCH] F: Mod.mainX

---
 core/src/mindustry/mod/Mods.java | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/core/src/mindustry/mod/Mods.java b/core/src/mindustry/mod/Mods.java
index 8d9c9b2d3e4807e989fa3462a08b3c00d1987042..f65efc3fd562850aaa72ea2170544d825ad7aeb2 100644
--- a/core/src/mindustry/mod/Mods.java
+++ b/core/src/mindustry/mod/Mods.java
@@ -23,6 +23,7 @@ import mindustry.graphics.MultiPacker.*;
 import mindustry.mod.ContentParser.*;
 import mindustry.type.*;
 import mindustry.ui.*;
+import mindustryX.*;
 import mindustryX.features.MetricCollector.*;
 
 import java.io.*;
@@ -1149,6 +1150,11 @@ public class Mods implements Loadable{
 
             String camelized = meta.name.replace(" ", "");
             String mainClass = meta.main == null ? camelized.toLowerCase(Locale.ROOT) + "." + camelized + "Mod" : meta.main;
+            //MDTX: prefer mainX as mainClass
+            if(meta.mainX != null){
+                Log.info("Mod @ uses mainX override: @", meta.name, meta.mainX);
+                mainClass = meta.mainX;
+            }
             String baseName = meta.name.toLowerCase(Locale.ROOT).replace(" ", "-");
 
             var other = mods.find(m -> m.name.equals(baseName));
@@ -1439,6 +1445,9 @@ public class Mods implements Loadable{
         /** Minimum game version that this mod requires, e.g. "140.1" */
         public String minGameVersion = "0";
         public @Nullable String displayName, author, description, subtitle, version, main, repo;
+        /** Main for MindustryX, has a higher priority than main.*/
+        @MindustryXApi
+        public @Nullable String mainX;
         public Seq<String> dependencies = Seq.with();
         public Seq<String> softDependencies = Seq.with();
         /** Hidden mods are only server-side or client-side, and do not support adding new content. */
